lib/webdriveragent.js

```diff
@@ -302,8 +373,15 @@ class WebDriverAgent {
     }
     this.log.info('Launching WebDriverAgent on the device without xcodebuild');
     if (this.isRealDevice) {
-      this.xctestApiClient = new Xctest(this.device.udid, this.bundleIdForXctest, null, {env: xctestEnv});
-      await this.xctestApiClient.start();
+      // Current method to launch WDA process can be done via 'xcrun devicectl',
+      // but it has limitation about the WDA preinstalled package.
+      // https://github.com/appium/appium/issues/19206#issuecomment-2014182674
+      if (util.compareVersions(this.platformVersion, '>=', '17.0')) {
+        await this._launchWdaViaDeviceCtl({env: xctestEnv});
```

Comment:
_launchViaDevicectl should be enough as the name of the class itself is WebDriverAgent